app/src/main/java/com/example/sharingang/users/FirestoreUserStore.kt

```diff
@@ -30,4 +35,22 @@ class FirestoreUserStore @Inject constructor(firestore: FirebaseFirestore) :
         return super.update(user, user.id)
     }
 
+    override suspend fun report(reportedUser: User, reporterUser: User): Boolean {
+        return try {
+            firestore
+                .collection("users").document(reportedUser.id!!)
+                .collection("reports").document(reporterUser.id!!).set(
+                    hashMapOf(
+                        reporterUser.id to Calendar.getInstance().time.toString()
```

Comment:
Will this create something like?
```json
{
  "reporter-id-4fe$f34lkd": "Apr 22 21:40:14"
}
```

If that's the case maybe it would be better to have
```json
{
  "reporter": "reporter-id-4fe$f34lkd",
  "reportedAt": "Apr 22 21:40:14"
}
```

Reply:
Okay, I will try!

Reply:
And I think that you can use `Date()` without converting it into a String.
Firestore can store it as a date